src/coreclr/vm/prestub.cpp

```diff
@@ -2377,6 +2384,15 @@ PCODE MethodDesc::DoPrestub(MethodTable *pDispatchingMT, CallerGCMode callerGCMo
     }
     else
     {
+#ifdef FEATURE_PORTABLE_ENTRYPOINTS
+        pCode = pStub->GetEntryPoint();
+        pStub->DecRef();
+
+        void* ilStubInterpData = PortableEntryPoint::GetInterpreterData(pCode);
+        _ASSERTE(ilStubInterpData != NULL);
+        SetInterpreterCode((InterpByteCodeStart*)ilStubInterpData);
+        SetCodeEntryPoint(pCode);
```

Comment:
Are both `SetInterpreterCode` and `SetCodeEntryPoint` needed? I would expect that you need only one.

Reply:
When I don't set interpreter code I am getting
```
ASSERT FAILED
        Expression: portableEntryPoint->HasNativeCode()
        Location:   line 28 in /Users/rodo/git/runtime-main/src/coreclr/vm/precode_portable.cpp
        Function:   GetActualCode
        Process:    42
Frame (InterpreterFrame): 0x4fee58
   0) System.Text.ValueStringBuilder::AppendFormatHelper, IR_051c
   1) System.IO.StreamWriter::WriteFormatHelper, IR_007c
   2) System.IO.StreamWriter::WriteLine, IR_004d
   3) System.Console::WriteLine, IR_0018
   4) Program::<<Main>$>g__TestCallingConvention0|0_0, IR_0023
   5) Program::<Main>$, IR_0011
```
and when I don't set code entry point, it triggers the 2 asserts at the end of this method.

Reply:
Ok